packages/esm-patient-clinical-view-app/src/hooks/usePeers.ts

```diff
@@ -0,0 +1,112 @@
+import { formatDate, openmrsFetch, parseDate, useConfig } from '@openmrs/esm-framework';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { ConfigObject } from '../config-schema';
+import { Peer, Person, Relationship } from '../types';
+function extractName(display: string) {
+  const pattern = /-\s*(.*)$/;
+  const match = display.match(pattern);
+  if (match && match.length > 1) {
+    return match[1].trim();
+  }
+  return display.trim();
+}
+
+function extractValue(display: string) {
+  const pattern = /=\s*(.*)$/;
+  const match = display.match(pattern);
+  if (match && match.length > 1) {
+    return match[1].trim();
+  }
+  return display.trim();
+}
+
+function extractAttributeData(person: Person, config: ConfigObject) {
+  return person.attributes.reduce<{
+    contact: string | null;
+  }>(
+    (prev, attr) => {
+      if (attr.attributeType.uuid === config.contactPersonAttributesUuid.telephone) {
+        return { ...prev, contact: attr.display ? extractValue(attr.display) : null };
+      }
+      return prev;
+    },
+    {
+      contact: null,
+    },
+  );
+}
+
+function getPeer(relationship: Relationship, config: ConfigObject, person: 'personA' | 'personB') {
+  return {
+    ...extractAttributeData(relationship[person], config),
+    uuid: relationship.uuid,
+    name: extractName(relationship[person].display),
+    display: relationship[person].display,
+    relativeAge: relationship[person].age,
+    dead: relationship[person].dead,
+    causeOfDeath: relationship[person].causeOfDeath,
+    relativeUuid: relationship[person].uuid,
+    relationshipType: relationship.relationshipType.bIsToA,
+    patientUuid: relationship[person].uuid,
+    gender: relationship[person].gender,
+    startDate: !relationship.startDate ? null : formatDate(parseDate(relationship.startDate)),
```

Comment:
If this is to be display on the UI, its better to have the empty placeholder e.g 

```suggestion
    startDate: !relationship.startDate ? '--' : formatDate(parseDate(relationship.startDate)),
```

instead of having it as null

Reply:
That is handled at the table level when passing the rows props